silence/silence.go

```diff
@@ -276,6 +325,11 @@ func (s *Silences) setSilence(sil *pb.Silence) error {
 	if err != nil {
 		return err
 	}
+
+	_, err = s.mc.Add(sil)
+	if err != nil {
+		return err
+	}
```

Comment:
I'd vote for entirely skipping the `Add` then as it is not happening in them same place of the general adding of silences, which is what threw me off initially. There's no good reason to pre-cache some silences and on-demand others.

Then we can just move `Add()` entirely into `Get`.